src/CattyTests/DataModel/UserListTest.swift

```diff
@@ -81,11 +81,110 @@ final class UserListTest: XCTestCase {
         let userListB = UserList(name: "userListB")
         let userListC = UserList(name: "userList")
 
-        userListA.value = NSMutableArray(array: [50, 51, "item"])
-        userListB.value = NSMutableArray(array: [50, 51, "item"])
-        userListC.value = NSMutableArray(array: [50, 51, "item"])
+        userListA.add(element: NSMutableArray(array: [50, 51, "item"]))
+        userListB.add(element: NSMutableArray(array: [50, 51, "item"]))
+        userListC.add(element: NSMutableArray(array: [50, 51, "item"]))
 
         XCTAssertFalse(userListB.isEqual(userListA))
         XCTAssertTrue(userListC.isEqual(userListA))
     }
+
+    func testAddToUserList() {
+        let list1 = UserList(name: "testName")
+
+        list1.add(element: 10)
+        list1.add(element: 20)
+
+        let value = list1.value
+
+        XCTAssertEqual(2, value.count)
+        XCTAssertEqual(value[0] as! Int, 10)
+        XCTAssertEqual(value[1] as! Int, 20)
+    }
+
+    func testDeleteFromUserList() {
+        let list1 = UserList(name: "testName")
+
+        list1.add(element: 10)
+        list1.add(element: 20)
+
+        var value = list1.value
+
+        XCTAssertEqual(2, value.count)
+
+        list1.delete(at: 2)
+        value = list1.value
+
+        XCTAssertEqual(1, value.count)
+        XCTAssertEqual(value[0] as! Int, 10)
+    }
+
+    func testInsertToUserList() {
+        let list1 = UserList(name: "testName")
+
+        list1.insert(element: 10, at: 1)
+        list1.insert(element: 30, at: 2)
+
+        var value = list1.value
+
+        XCTAssertEqual(value.count, 2)
+        XCTAssertEqual(value[0] as! Int, 10)
+        XCTAssertEqual(value[1] as! Int, 30)
+
+        list1.insert(element: 20, at: 3)
+
+        value = list1.value
+
+        XCTAssertEqual(value.count, 3)
+        XCTAssertEqual(value[0] as! Int, 10)
+        XCTAssertEqual(value[1] as! Int, 30)
+        XCTAssertEqual(value[2] as! Int, 20)
+    }
+
+    func testInsertToUserListInvalidIndex() {
+        let list1 = UserList(name: "testName1")
+
+        list1.insert(element: 10, at: -1)
+        var value = list1.value
+
+        XCTAssertEqual(value.count, 0)
+
+        list1.insert(element: 20, at: 2)
+        value = list1.value
+
+        XCTAssertEqual(value.count, 0)
+    }
+
+    func testReplaceItemInUserList() {
+        let list1 = UserList(name: "testName1")
+
+        list1.insert(element: 10, at: 1)
+        list1.insert(element: 20, at: 2)
+        list1.replace(at: 2, with: 30)
+
+        var value = list1.value
+
+        XCTAssertEqual(value.count, 2)
+        XCTAssertEqual(value[0] as! Int, 10)
+        XCTAssertEqual(value[1] as! Int, 30)
+
+        list1.replace(at: 1, with: 40)
+
+        value = list1.value
+
+        XCTAssertEqual(value.count, 2)
+        XCTAssertEqual(value[0] as! Int, 40)
+        XCTAssertEqual(value[1] as! Int, 30)
+    }
+
+    func testReplaceItemInUserListInvalidIndex() {
```

Comment:
Rename to `testReplaceInvalidIndex` since we are already in the `UserListTests` scope.